server/src/main/java/org/elasticsearch/transport/RemoteConnectionStrategy.java

```diff
@@ -178,17 +178,6 @@ static RemoteConnectionStrategy buildStrategy(String clusterAlias,
         }
     }
 
-    public static boolean isConnectionEnabled(Settings connectionSettings) {
```

Comment:
Didn't like the implicit removal of connections if the settings change to have an empty hosts list. I'd rather we only remove connections if subscriptions are dropped, and enforce that active subscriptions do need non-empty hosts.